# t-tests pvalues

One tailed t-test 

In [1]:
import pandas as pd
import numpy as np
from scipy.stats import ttest_ind

In [23]:
machine = pd.read_csv('files_for_lab/machine_new.txt', sep='\t')

In [30]:
machine.columns

Index(['New machine', '    Old machine'], dtype='object')

In [32]:
machine = machine.rename(columns={'New machine': 'new_machine', '    Old machine': 'old_machine'})

In [33]:
machine

new_machine  old_machine
0         42.1         42.7
1         41.0         43.6
2         41.3         43.8
3         41.8         43.3
4         42.4         42.5
5         42.8         43.5
6         43.2         43.1
7         42.3         41.7
8         41.8         44.0
9         42.7         44.1

In [34]:
t_stat, p_val = ttest_ind(machine['new_machine'], machine['old_machine'], alternative='less')

if p_val < 0.05:
    print("The data provides sufficient evidence to show that the new machine packs cartons faster on average than the old machine.")
else:
    print("The data does not provide sufficient evidence to show that the new machine packs cartons faster on average than the old machine.")

The data provides sufficient evidence to show that the new machine packs cartons faster on average than the old machine.


Matched Pairs Test

In [17]:
from scipy.stats import ttest_rel

In [18]:
pokemon = pd.read_csv('files_for_lab/pokemon.csv')

In [19]:
# perform the test
t_statistic, p_value = ttest_rel(pokemon['Attack'], pokemon['Defense'])

In [21]:
# the results
print("t-statistic: ", t_statistic)
print("p-value: ", p_value)

t-statistic:  4.325566393330478
p-value:  1.7140303479358558e-05


The t-statistic is 4.33, and the p-value is very small (1.71e-05), which suggests strong evidence against the null hypothesis. Therefore, we can reject the null hypothesis and conclude that there is a significant difference between attack and defense scores of the Pokemon.